source/geod24/LocalRest.d

```diff
@@ -400,11 +400,64 @@ public final class RemoteAPI (API) : API
 
     private static void spawned (Implementation) (CtorParams!Implementation cargs)
     {
+        import std.datetime.systime : Clock, SysTime;
+        import std.algorithm;
+        import std.range;
+
         scope node = new Implementation(cargs);
         scheduler = new LocalScheduler;
         scope exc = new Exception("You should never see this exception - please report a bug");
 
+        // very simple & limited variant, to keep it performant.
+        // should be replaced by a real Variant later
+        static struct Variant
+        {
+            this (Response res) { this.res = res; this.tag = 0; }
+            this (Command cmd) { this.cmd = cmd; this.tag = 1; }
+
+            union
+            {
+                Response res;
+                Command cmd;
+            }
+
+            ubyte tag;
+        }
+
+        // can't embed SysTime in TimeCommand as it has internal aliasing
+        struct Sleep
```

Comment:
What I had in mind was more of a `Control` struct where `filter` would be stored as well. But not a big deal.